src/traits.rs

```diff
@@ -1251,23 +1251,15 @@ impl NomRange<usize> for Range<usize> {
   }
 
   fn is_inverted(&self) -> bool {
-    !(self.start < self.end)
+    self.is_empty()
   }
 
   fn saturating_iter(&self) -> Self::Saturating {
-    if self.end == 0 {
-      1..0
-    } else {
-      0..self.end - 1
-    }
+    0..self.end - 1
```

Comment:
Wouldn't this panic when overflow checking is enabled (such as when building with debug)?
And even if the underflow does not panic it would return `0..usize::MAX` as the iterator for a range of `x..0`. This is incorrect because the range should be empty. Similar reasoning applies to the other range implementations.